src/sentry/models/organization.py

```diff
@@ -164,10 +164,16 @@ def save(self, *args, **kwargs):
         else:
             super().save(*args, **kwargs)
 
-    def delete(self):
+    def delete(self, **kwargs):
+        from sentry.models import NotificationSetting
+
         if self.is_default:
             raise Exception("You cannot delete the the default organization.")
-        return super().delete()
+
+        # There is no foreign key relationship so we have to manually cascade.
+        NotificationSetting.objects.remove_for_organization(self)
```

Comment:
This and some other files seem to have duplicated code changes that are in https://github.com/getsentry/sentry/pull/24360/